src/index.js

```diff
@@ -160,8 +161,10 @@ class Service {
   }
 
   _create(data, params) {
-    return this.db().insert(data, this.id).then(rows => this._get(rows[0], params))
-      .catch(errorHandler);
+    return this.db().insert(data, this.id).then(rows => {
+      const id = this.idFromData ? data[this.id] : rows[0];
```

Comment:
Does it need to be an option flag? I feel we could just fairly safely assume `const id = typeof data[this.id] !== 'undefined' ? data[this.id] : rows[0]`?